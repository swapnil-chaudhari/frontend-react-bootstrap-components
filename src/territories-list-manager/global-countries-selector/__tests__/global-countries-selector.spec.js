import React from 'react';
import renderShallow from 'render-shallow';
import { findWithClass } from 'react-shallow-testutils';
import GlobalountriesSelector from '../global-countries-selector';

describe('<GlobalountriesSelector>', () => {
    const formatMessage = msg => msg;
    const messages = {
        selectAllLabel: 'select all label',
        selectNoneLabel: 'select none label'
    };

    describe('when it renders', () => {
        let component;
        const props = {
            onTerritorySelect: jest.fn(),
            formatMessage,
            messages
        };

        beforeAll(() => {
            component = renderShallow(<GlobalountriesSelector { ...props } />).output;
        });

        test('should render correct class', () => {
            expect(component.props.className).toEqual('GlobalCountriesSelector');
        });

        test('should render select all and select none elements', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when select all clicked', () => {
        let component;
        const props = {
            onTerritorySelect: jest.fn(),
            formatMessage,
            messages
        };

        beforeAll(() => {
            component = renderShallow(<GlobalountriesSelector { ...props } />).output;
            findWithClass(component, 'GlobalCountriesSelector-all btn-link').props.onClick();
        });

        test('should call onTerritorySelect with all param', () => {
            expect(props.onTerritorySelect).toHaveBeenCalledWith(
                null,
                true
            );
        });
    });

    describe('when select none clicked', () => {
        let component;
        const props = {
            onTerritorySelect: jest.fn(),
            formatMessage,
            messages
        };

        beforeAll(() => {
            component = renderShallow(<GlobalountriesSelector { ...props } />).output;
            findWithClass(component, 'GlobalCountriesSelector-none btn-link').props.onClick();
        });

        test('should call onTerritorySelect with none param', () => {
            expect(props.onTerritorySelect).toHaveBeenCalledWith(
                null,
                null,
                true
            );
        });
    });
});
