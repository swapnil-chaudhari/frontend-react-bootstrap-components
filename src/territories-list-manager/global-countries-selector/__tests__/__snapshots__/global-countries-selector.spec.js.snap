// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalountriesSelector> when it renders should render select all and select none elements 1`] = `
<div
  className="GlobalCountriesSelector"
>
  <b>
    Select:
  </b>
  <span
    className="GlobalCountriesSelector-all btn-link"
    onClick={[Function]}
  >
    <b>
      select all label
    </b>
  </span>
  <span
    className="GlobalCountriesSelector-none btn-link"
    onClick={[Function]}
  >
    <b>
      select none label
    </b>
  </span>
</div>
`;
